jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackEncoder.java

```diff
@@ -180,26 +176,30 @@ else if (metadata.isResponse())
             encode(buffer, status);
         }
 
-        // Add all non-connection fields.
+        // Remove fields as specified in RFC 7540, 8.1.2.2.
         HttpFields fields = metadata.getFields();
         if (fields != null)
         {
-            Set<String> hopHeaders = fields.getCSV(HttpHeader.CONNECTION, false).stream()
-                .filter(v -> specialHopHeaders.get(v) == Boolean.TRUE)
-                .map(StringUtil::asciiToLowerCase)
-                .collect(Collectors.toSet());
+            // For example: Connection: Close, TE, Upgrade, Custom.
+            List<String> values = fields.getCSV(HttpHeader.CONNECTION, false);
+            Set<String> hopHeaders = new HashSet<>();
```

Comment:
I wonder if it is worthwhile creating this set.   The Connection field is not that frequently used and when it is, has few entries.   It is probably better just to have the list, save creating the empty hashset and live with the slightly less efficient contains method (probably not so bad for lists of only a few items).

Reply:
> I wonder if it is worthwhile creating this set. The Connection field is not that frequently used and when it is, has few entries. It is probably better just to have the list, save creating the empty hashset and live with the slightly less efficient contains method (probably not so bad for lists of only a few items).

This makes sense

Reply:
+1